src/Files.App/ViewModels/Properties/SecurityViewModel.cs

```diff
@@ -3,255 +3,195 @@
+						: System.Security.AccessControl.InheritanceFlags.None,
+					IsInherited = false,
+					PrincipalSid = pickedSid,
+					PropagationFlags = System.Security.AccessControl.PropagationFlags.None,
+				};
+
+				AccessControlList.AccessControlEntryPrimitiveMappings.Add(mapping);
+				AccessControlList.AccessControlEntries.Add(new(mapping, IsFolder));
+
+				SaveChangedAccessControlList();
 			}
 		}
 
-		private async void EditOwner()
+		private void RemoveAccessControlEntry()
 		{
-			var pickedObject = await OpenObjectPicker();
-			if (pickedObject is not null)
-			{
-				// Refresh file permissions
-				if (SetFileOwner(pickedObject))
-					GetFilePermissions();
-			}
+			AccessControlList.AccessControlEntryPrimitiveMappings.RemoveAll(x =>
+				x.AccessControlType == (System.Security.AccessControl.AccessControlType)SelectedAccessControlEntry.AccessControlType &&
+				x.FileSystemRights == (System.Security.AccessControl.FileSystemRights)SelectedAccessControlEntry.AccessMaskFlags &&
+				x.InheritanceFlags == (System.Security.AccessControl.InheritanceFlags)SelectedAccessControlEntry.InheritanceFlags &&
+				x.IsInherited == SelectedAccessControlEntry.IsInherited &&
+				x.PrincipalSid == SelectedAccessControlEntry.PrincipalSid &&
+				x.PropagationFlags == (System.Security.AccessControl.PropagationFlags)SelectedAccessControlEntry.PropagationFlags);
+			AccessControlList.AccessControlEntries.Remove(SelectedAccessControlEntry);
+
+			SaveChangedAccessControlList();
 		}
 
-		private async void AddRulesForUser()
+		private void DisableInheritance()
 		{
-			var pickedObject = await OpenObjectPicker();
-			if (pickedObject is not null)
-			{
-				if (!FilePermissions.RulesForUsers.Any(x => x.UserGroup.Sid == pickedObject))
-				{
-					// No existing rules, add user to list
-					FilePermissions.RulesForUsers.Add(RulesForUser.ForUser(FilePermissions.AccessRules, IsFolder, pickedObject));
-				}
-			}
+			// Update protection status and refresh access control
+			if (FileOperationsHelpers.SetAccessRuleProtection(Item.ItemPath, IsFolder, _isProtected, _preserveInheritance))
+				GetAccessControlList();
 		}
 
-		private void RemoveRulesForUser()
+		private void SetDisableInheritanceOption(string options)
 		{
-			if (SelectedRuleForUser is not null)
-			{
-				SelectedRuleForUser.AllowRights = 0;
-				SelectedRuleForUser.DenyRights = 0;
+			_isProtected = bool.Parse(options.Split(',')[0]);
+			_preserveInheritance = bool.Parse(options.Split(',')[1]);
 
-				if (!FilePermissions.AccessRules.Any(x => x.IdentityReference == SelectedRuleForUser.UserGroup.Sid))
-				{
-					// No remaining rules, remove user from list
-					FilePermissions.RulesForUsers.Remove(SelectedRuleForUser);
-				}
-			}
+			OnPropertyChanged(nameof(DisableInheritanceOption));
+			DisableInheritanceCommand.NotifyCanExecuteChanged();
 		}
 
-		public void GetFilePermissions()
+		private void ReplaceChildPermissions()
```

Comment:
This method is empty. Why keep it?

Reply:
This had already been empty before having worked on refactoring.

Reply:
I can add this. But we should limit the scope of this PR.